tests/foreman/ui/test_ldap_authentication.py

```diff
@@ -763,3 +803,68 @@ def test_positive_login_ad_user_basic_roles(
             ldapsession.usergroup.search('')
         ldapsession.architecture.create({'name': name})
         assert ldapsession.architecture.search(name)[0]['Name'] == name
+
+
+@upgrade
+@tier2
+def test_positive_login_user_password_otp(test_name, ipa_data, auth_source_ipa):
+    """Login with password with time based OTP
+
+    :id: be7eb5d6-3228-4660-aa64-c56f9f3ec5e0
+
+    :setup: Assure properly functioning IPA server for authentication
+
+    :steps: Login to server with an IPA user with time_based OTP.
+
+    :expectedresults: Log in to foreman UI successfully
+
+    """
+    try:
+        password_with_otp = "{0}{1}".format(
+            ipa_data['ldap_ipa_user_passwd'],
+            generate_otp(ipa_data['time_based_secret']))
+        with Session(
+                test_name,
+                ipa_data['ipa_otp_username'],
+                password_with_otp
+        ) as ldapsession:
+            with raises(NavigationTriesExceeded):
+                ldapsession.user.search('')
+            expected_user = "{} {}".format(ipa_data['ipa_otp_username'],
+                                           ipa_data['ipa_otp_username'])
+            assert ldapsession.task.read_all()['current_user'] == expected_user
+        users = entities.User().search(query={
+            'search': 'login="{}"'.format(ipa_data['ipa_otp_username'])
+        })
+        assert users[0].login == ipa_data['ipa_otp_username']
+    finally:
+        ldap_tear_down()
+
+
+@tier2
+def test_negative_login_user_with_invalid_password_otp(test_name, ipa_data, auth_source_ipa):
+    """Login with password with time based OTP
+
+    :id: 3718c86e-5976-4fb8-9c80-4685d53bd955
+
+    :setup: Assure properly functioning IPA server for authentication
+
+    :steps: Login to server with an IPA user with invalid OTP.
+
+    :expectedresults: Log in to foreman UI should be failed
+
+    """
+    try:
+        password_with_otp = "{0}{1}".format(
+            ipa_data['ldap_ipa_user_passwd'],
+            gen_string(str_type='numeric', length=6))
+        with Session(
+                test_name,
+                ipa_data['ipa_otp_username'],
+                password_with_otp
+        ) as ldapsession:
+            with raises(NavigationTriesExceeded) as error:
+                ldapsession.user.search('')
+            assert error.typename == "NavigationTriesExceeded"
```

Comment:
This assertion is already done by [`pytest.raises`](https://docs.pytest.org/en/latest/reference.html#pytest-raises). If block within context raises different exception than specified, or does not raise exception at all, pytest will fail the test.

Reply:
That's okay right, it should only lead same error otherwise there has to be a problem with the user able to login or suppress.